teuthology/misc.py

```diff
@@ -911,6 +911,9 @@ def wait_until_osds_up(ctx, cluster, remote, ceph_cluster='ceph'):
     testdir = get_testdir(ctx)
     with safe_while(sleep=6, tries=50) as proceed:
         while proceed():
+            daemons = ctx.daemons.iter_daemons_of_role('osd', ceph_cluster)
```

Comment:
maybe we also want to do something similar in wait_until_healthy, perhaps with more daemon types (don't remember if non-osds are also started without waiting)?  Can be a followon, but seems similarly useful